vcl_templates/www.vcl.erb

```diff
@@ -532,26 +532,17 @@ sub vcl_deliver {
     add resp.http.Set-Cookie = "ABTest-Example=" req.http.GOVUK-ABTest-Example "; secure; expires=" now + 1d;
   }
 
-  # Hard coded test to check things are working ok because so far things aren't
-  if (req.url ~ "^/bank-holidays"
-    && req.http.User-Agent !~ "^GOV\.UK Crawler Worker"
-    && req.http.Cookie !~ "ABTest-BankHolidaysTest") {
-    add resp.http.Set-Cookie = "ABTest-BankHolidaysTest=" req.http.GOVUK-ABTest-BankHolidaysTest "; secure; expires=" now + 1d;
-  }
-
   # Begin dynamic section
 <% if ab_tests -%>
   declare local var.expiry TIME;
 <% ab_tests.each do |test_config| -%>
 <% test_config.each do |test, _| -%>
 <% unless test == "Example" -%>
-  if (req.http.Cookie ~ "cookies_policy") {
-    if (req.http.Cookie:cookies_policy ~ "%22usage%22:true") {
-      if (table.lookup(active_ab_tests, "<%= test %>") == "true") {
-        if (req.http.Cookie !~ "ABTest-<%= test %>" || req.url ~ "[\?\&]ABTest-<%= test %>" && req.http.User-Agent !~ "^GOV\.UK Crawler Worker") {
-          set var.expiry = time.add(now, std.integer2time(std.atoi(table.lookup(ab_test_expiries, "<%= test %>"))));
-          add resp.http.Set-Cookie = "ABTest-<%= test %>=" req.http.GOVUK-ABTest-<%= test %> "; secure; expires=" var.expiry "; path=/";
-        }
+  if (req.http.Usage-Cookies-Opt-In ~ "true") {
+    if (table.lookup(active_ab_tests, "<%= test %>") == "true") {
+      if (req.http.Cookie !~ "ABTest-<%= test %>" || req.url ~ "[\?\&]ABTest-<%= test %>" && req.http.User-Agent !~ "^GOV\.UK Crawler Worker") {
```

Comment:
As you mentioned in the commit message, `req.http.Cookie` is always going to be `(null)` here, so the first bit of this conditional is always true (since `(null)` doesn't regex match "ABTest-blah").

I think we could fix this by using the `req.http.GOVUK-ABTest-<%= test %>` header, which will be `(null)` (or possibly empty string), unless the cookie was set, in which case it will be "A" or "B" etc. [std.strlen](https://developer.fastly.com/reference/vcl/functions/strings/std-strlen/) seems to be the recommended way of checking either null or empty in VCL.

```suggestion
      if (std.strlen(req.http.GOVUK-ABTest-<%= test %>) == 0 || req.url ~ "[\?\&]ABTest-<%= test %>" && req.http.User-Agent !~ "^GOV\.UK Crawler Worker") {
```

Reply:
Turns out `req.http.GOVUK-ABTest-...` gets set even when the user hasn't got a cookie yet, so we can't use that for this check. Kelvin and I paired on this and added another header which gets set if the cookie is already present in e750728, which should fix this.